lib/components/viewers/route-viewer.js

```diff
@@ -76,13 +84,17 @@ class RouteViewer extends Component {
           {agencySortedRoutes
             .map(route => {
               // Find operator based on agency_id (extracted from OTP route ID).
-              const operator = operatorForRoute(operators, route) || {}
```

Comment:
@landonreed, this is where the problem was. On each re-render of the RouteViewer, the operatorForRoute will return a new `{}`. Each of these new `{}` objects does not strictly equal the other `{}` instances due to [the way JavaScript calculates object equality](http://adripofjavascript.com/blog/drips/object-equality-in-javascript.html), therefore the props are different which causes the rerenders.

Reply:
Gotcha, thanks for fixing. Before I approve this though, I think we should address the fact that a re-render occurs every time a new pattern geometry response is processed and added to the route in the reducer. Whether this fix happens by deconstructing the route into its parts and passing them piecemeal to the `RouteRow` component or through some change in the reducer, I don't think this PR will have achieved its objective until those extra renders are fixed.

Reply:
Yes, that would further optimize things, but that is a separate optimization entirely. This refactor merely touches one file. The primary objective of ensuring that each route row doesn't re-render when a change occurs to an unrelated route row has already been accomplished within this PR.

If I were to attempt to implement the pattern geometry optimization, I'd edit the api actions so that a new action wouldn't get dispatched until all of the geometries for the pattern were fetched.

Reply:
Fair enough. Can you create a ticket describing that approach? After that's done feel free to merge.

Reply:
Added a comment to #64.